### PR TITLE
tested and linted snippets

### PR DESCRIPTION
**Pull request by sh-rp**

### CODE REVIEW

**Review by rudolfix — CHANGES_REQUESTED** [review]
this is really good. thanks for taking this task :)

now we need to decouple snippet tests and linting from the core CI. (maybe it is time to move docosaurus out?)
* we want the docs changes to run quickly on CI - like we do it now
* we should have separate linting and testing workflow for snippets and examples. it should run only if anything in docs changed!
* let's keep them in docs folder to make them isolated

File: docs/website/docs/intro.md [hunk 1]
```diff
@@ -19,25 +20,28 @@ then `import dlt`. Once set up, it will automatically load any
...
+    )
+    # Grab some player data from Chess.com API
+    data = []
+    for player in ['magnuscarlsen', 'rpragchess']:
+        response = requests.get(f'https://api.chess.com/pub/player/{player}')
+        response.raise_for_status()
+        data.append(response.json())
+    # Extract, normalize, and load the data
+    info = pipeline.run(data, table_name='player')
+    print(info)
```
Comment (AstrakhantsevaAA) [inline docs/website/docs/intro.md:42]: can you delete extra tabs? it looks weird now(
Снимок экрана 2023-08-21 в 12 07 13

Comment (AstrakhantsevaAA) [reply docs/website/docs/intro.md:42]: before:
Снимок экрана 2023-08-21 в 12 08 06

Comment (sh-rp) [reply docs/website/docs/intro.md:42]: please see the comments about this on slack

**Review by rudolfix — APPROVED** [review]
heh `exec` worker. Python is scary :)

File: docs/snippets/utils.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from typing import Dict, Any
+
+BASEPATH = "docs/snippets"
+
+def run_snippet(filename: str) -> Dict[str, Any]:
+    variables: Dict[str, Any] = {}
+    exec(open(f"{BASEPATH}/{filename}.py", encoding="utf-8").read(), variables)
```
Comment (rudolfix) [inline docs/snippets/utils.py:7]: change the working dir to base path and use os.path.join to merge paths

**Review by rudolfix — CHANGES_REQUESTED** [review]

File: docs/snippets/intro_snippet_test.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+
+from tests.pipeline.utils import assert_load_info
+from docs.snippets.utils import run_snippet
```
Comment (rudolfix) [inline docs/snippets/intro_snippet_test.py:3]: add fixtures: preserve_environ, autouse_test_storage, patch_home_dir so the tests are isolated

**Review by rudolfix — APPROVED** [review]

**Pull request merged**